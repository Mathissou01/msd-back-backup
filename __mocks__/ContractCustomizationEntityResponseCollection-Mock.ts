export const contractCustomizationEntityResponseCollectionMock = {
  data: [
    {
      attributes: {
        homepage: {
          data: {
            attributes: {
              quizAndTipsBlock: {
                data: {
                  id: "1",
                  attributes: {
                    title: "Quiz & Astuces",
                    displayBlock: false,
                    displayQuiz: false,
                    quiz: {
                      data: {
                        id: "2",
                        attributes: {
                          title: "Nom du quiz",
                          publishedAt: "2022-11-20T02:34:14.535Z",
                        },
                      },
                    },
                    displayTips: false,
                    tips: {
                      data: [],
                    },
                  },
                },
              },
            },
          },
        },
      },
    },
  ],
};
