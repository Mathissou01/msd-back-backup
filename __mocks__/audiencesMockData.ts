import { GetAudiencesByContractIdDocument } from "../src/graphql/codegen/generated-types";

export const getNoAudienceMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [],
        },
      },
    },
  },
];

export const getSingleAudienceMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];

export const getTwoAudiencesMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
            {
              id: 2,
              attributes: {
                isActive: true,
                type: "Collectifs",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];

export const getAllAudiencesMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
            {
              id: 2,
              attributes: {
                isActive: true,
                type: "Collectifs",
                __typename: "Audience",
              },
            },
            {
              id: 3,
              attributes: {
                isActive: true,
                type: "Professionnels",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];
