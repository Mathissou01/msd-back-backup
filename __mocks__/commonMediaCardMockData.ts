import { GetFilesPaginationByFolderIdDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetFilesPaginationByFolderIdDocument,
      filters: {
        folder: {
          id: {
            eq: 2,
          },
        },
      },
      pagination: {
        page: 1,
        pageSize: 10,
      },
      sort: "createdAt:desc",
    },
    result: {
      data: {
        uploadFiles: {
          data: [
            {
              id: "124",
              attributes: {
                name: "La Bruja.jpeg",
                mime: "image/jpeg",
                size: 124.35,
                width: 1280,
                height: 853,
                createdAt: "2023-01-26T10:34:26.058Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/La%20Bruja.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "122",
              attributes: {
                name: "Banana Mama Rooftop & Kitchen 5.jpeg",
                mime: "image/jpeg",
                size: 68.84,
                width: 960,
                height: 960,
                createdAt: "2023-01-26T10:30:22.701Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Banana%20Mama%20Rooftop%20%26%20Kitchen%205.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "121",
              attributes: {
                name: "Sichuan Hot Pot.jpeg",
                mime: "image/jpeg",
                size: 2233.13,
                width: 6962,
                height: 4615,
                createdAt: "2023-01-18T17:39:30.518Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Sichuan%20Hot%20Pot.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "120",
              attributes: {
                name: "mahbod-akhzami-kg40JrEsCvY-unsplash.jpg",
                mime: "image/jpeg",
                size: 1788.48,
                width: 6000,
                height: 4000,
                createdAt: "2023-01-18T17:39:29.900Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/mahbod-akhzami-kg40JrEsCvY-unsplash.jpg",
                ext: ".jpg",
              },
            },
            {
              id: "119",
              attributes: {
                name: "Paella in Valencia.jpeg",
                mime: "image/jpeg",
                size: 1026.93,
                width: 3851,
                height: 2528,
                createdAt: "2023-01-18T17:39:29.720Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Paella%20in%20Valencia.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "118",
              attributes: {
                name: "Rusconi's American Kitchen 3.jpeg",
                mime: "image/jpeg",
                size: 209.81,
                width: 1999,
                height: 1333,
                createdAt: "2023-01-18T17:39:29.502Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Rusconi%27s%20American%20Kitchen%203.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "117",
              attributes: {
                name: "Quince 5.jpeg",
                mime: "image/jpeg",
                size: 197.32,
                width: 2000,
                height: 1500,
                createdAt: "2023-01-18T17:39:29.310Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Quince%205.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "116",
              attributes: {
                name: "Rusconi's American Kitchen 2.jpeg",
                mime: "image/jpeg",
                size: 72.49,
                width: 1000,
                height: 1500,
                createdAt: "2023-01-18T17:39:29.221Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Rusconi%27s%20American%20Kitchen%202.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "115",
              attributes: {
                name: "Pur' - Jean-Fran√ßois Rouquette 3.jpeg",
                mime: "image/jpeg",
                size: 63.2,
                width: 1280,
                height: 853,
                createdAt: "2023-01-18T17:39:29.039Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Pur%27%20-%20Jean-Fran%C3%A7ois%20Rouquette%203.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "114",
              attributes: {
                name: "muneeb-muhamed-kPXT9HA4LOk-unsplash.jpg",
                mime: "image/jpeg",
                size: 516.64,
                width: 3456,
                height: 4608,
                createdAt: "2023-01-18T17:39:28.922Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/muneeb-muhamed-kPXT9HA4LOk-unsplash.jpg",
                ext: ".jpg",
              },
            },
          ],
          meta: {
            pagination: {
              total: 122,
              pageSize: 10,
              pageCount: 13,
              page: 1,
            },
          },
        },
      },
    },
  },
];
