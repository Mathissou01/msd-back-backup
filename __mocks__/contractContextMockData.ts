import {
  Enum_Contract_Clienttype,
  Enum_Contract_Contractstatus,
} from "../src/graphql/codegen/generated-types";

export const mockData = {
  id: "1",
  attributes: {
    clientName: "Grand test",
    clientType: "city" as Enum_Contract_Clienttype,
    contractStatus: "Initialisation" as Enum_Contract_Contractstatus,
    siret: 12345678910123,
    clear: 123456789,
    ccap: 123456789,
    isNonExclusive: false,
    isRVFrance: true,
    pathId: 2,
    logo: {
      data: null,
    },
    channelType: {
      data: {
        id: "1",
        attributes: {
          hasWebApp: false,
          hasWebSite: false,
        },
      },
    },
    clientContact: {
      data: {
        id: "1",
        attributes: {
          firstName: "test",
          lastName: "test",
          email: "test.test@mail.com",
          phoneNumber: "07 81 21 56 81",
        },
      },
    },
    contractCustomization: {
      data: {
        id: "1",
        attributes: {
          primaryColor: "#9BCD41",
          secondaryColor: "#FFC229",
          textContrast: "#030F40",
        },
      },
    },
    editorialService: {
      data: {
        id: "1",
        attributes: {
          eventSubService: {
            data: {
              id: "1",
              attributes: {
                name: "Evénement ",
                isActivated: true,
              },
            },
          },
          freeContentSubServices: {
            data: [
              {
                id: "1",
                attributes: {
                  name: "Editorial 1",
                  isActivated: true,
                },
              },
              {
                id: "2",
                attributes: {
                  name: "Editorial 2",
                  isActivated: true,
                },
              },
            ],
          },
          newsSubService: {
            data: {
              id: "1",
              attributes: {
                name: "Actualités",
                isActivated: true,
              },
            },
          },
          quizSubService: {
            data: {
              id: "1",
              attributes: {
                name: "Quiz",
                isActivated: true,
              },
            },
          },
          tipSubService: {
            data: {
              id: "1",
              attributes: {
                name: "Astuces",
                isActivated: true,
              },
            },
          },
        },
      },
    },
    recyclingGuideService: {
      data: {
        id: "1",
        attributes: {
          name: "Guide du tri",
          isActivated: false,
          memoName: "Memo Tri",
        },
      },
    },
    pickUpDayService: {
      data: {
        id: "1",
        attributes: {
          name: "Jour de collecte",
          isActivated: false,
        },
      },
    },
    dropOffMapService: {
      data: {
        id: "1",
        attributes: {
          name: "Points d'apports et carte",
          isActivated: false,
        },
      },
    },
    requestService: {
      data: {
        id: "1",
        attributes: {
          name: "Demande",
          isActivated: false,
        },
      },
    },
  },
};
