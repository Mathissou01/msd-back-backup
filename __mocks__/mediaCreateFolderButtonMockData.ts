import { GetAllFoldersHierarchyDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetAllFoldersHierarchyDocument,
      variables: {
        path: "/1",
        pathId: "6",
      },
    },
    result: {
      data: {
        getAllFoldersHierarchy: [
          {
            id: "1",
            name: "Contracts",
            path: "/1",
            pathId: "1",
          },
          {
            id: "6",
            name: "contract2",
            path: "/1/6",
            pathId: "6",
          },
          {
            id: "32",
            name: "hello",
            path: "/1/26",
            pathId: "26",
          },
          {
            id: "16",
            name: "dede",
            path: "/1/6/12",
            pathId: "12",
          },
          {
            id: "37",
            name: "xfgcfhf,h,ut",
            path: "/1/26/29",
            pathId: "29",
          },
          {
            id: "41",
            name: "lulu",
            path: "/1/6/33",
            pathId: "33",
          },
          {
            id: "38",
            name: "cghcg,h,cgh",
            path: "/1/26/29/30",
            pathId: "30",
          },
          {
            id: "42",
            name: "dedechildren2",
            path: "/1/6/12/34",
            pathId: "34",
          },
          {
            id: "43",
            name: "dedechildren1",
            path: "/1/6/12/35",
            pathId: "35",
          },
          {
            id: "45",
            name: "dedechildren3",
            path: "/1/6/12/37",
            pathId: "37",
          },
          {
            id: "39",
            name: "fgnxfchgf,c",
            path: "/1/26/29/30/31",
            pathId: "31",
          },
          {
            id: "44",
            name: "dedechildrenchildren1",
            path: "/1/6/12/35/36",
            pathId: "36",
          },
        ],
      },
    },
  },
];
