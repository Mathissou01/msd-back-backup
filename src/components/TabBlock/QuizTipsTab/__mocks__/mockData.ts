import { GetQuizAndTipsBlockDocument } from "../../../../graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetQuizAndTipsBlockDocument,
      variables: {
        contractId: "1",
      },
    },
    result: {
      data: {
        contractCustomizations: {
          data: [
            {
              attributes: {
                homepage: {
                  data: {
                    attributes: {
                      quizAndTipsBlock: {
                        data: {
                          id: "1",
                          attributes: {
                            title: "Titre modifi√©",
                            displayBlock: false,
                            displayQuiz: false,
                            quiz: {
                              data: {
                                id: "2",
                                attributes: {
                                  title: "Nom du quiz",
                                  publishedAt: "2022-11-20T02:34:14.535Z",
                                },
                              },
                            },
                            displayTips: false,
                            tips: {
                              data: [],
                            },
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          ],
        },
        services: {
          data: [
            {
              attributes: {
                serviceInstance: [
                  {
                    __typename: "ComponentMsdEditorial",
                    id: "4",
                    editorialServices: {
                      data: [
                        {
                          attributes: {
                            subServiceInstance: [
                              {
                                __typename: "ComponentEditoQuizzesSubService",
                                isActivated: true,
                                quizzes: {
                                  data: [
                                    {
                                      id: "2",
                                      attributes: {
                                        title: "Nom du quiz",
                                        publishedAt: "2022-11-20T02:34:14.535Z",
                                      },
                                    },
                                    {
                                      id: "1",
                                      attributes: {
                                        title: "truc",
                                        publishedAt: "2022-11-18T10:16:08.936Z",
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                        {
                          attributes: {
                            subServiceInstance: [
                              {
                                __typename: "ComponentEditoTipsSubService",
                                isActivated: true,
                                tips: {
                                  data: [
                                    {
                                      id: "5",
                                      attributes: {
                                        title: "nah",
                                        publishedAt: "2022-11-19T00:14:37.403Z",
                                      },
                                    },
                                    {
                                      id: "2",
                                      attributes: {
                                        title: "Titre de l'astuce 1",
                                        publishedAt: "2022-11-19T00:14:08.232Z",
                                      },
                                    },
                                    {
                                      id: "1",
                                      attributes: {
                                        title: "Titre de l'astuce 2",
                                        publishedAt: "2022-11-18T10:17:11.158Z",
                                      },
                                    },
                                    {
                                      id: "3",
                                      attributes: {
                                        title: "Titre de l'astuce 3",
                                        publishedAt: "2022-11-19T00:14:20.490Z",
                                      },
                                    },
                                    {
                                      id: "4",
                                      attributes: {
                                        title: "Titre de l'astuce 4",
                                        publishedAt: "2022-11-19T00:14:30.273Z",
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
            },
          ],
        },
      },
    },
  },
];

export const errorMockData = [
  {
    request: {
      query: GetQuizAndTipsBlockDocument,
      variables: {
        contractId: "1",
      },
    },
    error: new Error("An error occured"),
  },
];
