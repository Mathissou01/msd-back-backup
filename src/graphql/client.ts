import {
  ApolloClient,
  InMemoryCache,
  NormalizedCacheObject,
} from "@apollo/client";

let client: ApolloClient<NormalizedCacheObject>;

if (process.env.NEXT_PUBLIC_MOCK === "true") {
  /* With .env variable NEXT_PUBLIC_MOCK=true, return mock graphql client with fake response data */
  client = (await import("./mockedClient")).default(1000);
} else {
  client = await new ApolloClient({
    uri: `${process.env.NEXT_PUBLIC_API_URL}/graphql`,
    cache: new InMemoryCache({
      typePolicies: {
        UploadFolder: {
          merge: true,
        },
        UploadFile: {
          merge: true,
        },
        Contract: {
          merge: true,
        },
        ContractCustomization: {
          merge: true,
        },
        New: {
          merge: true,
        },
        Tip: {
          merge: true,
        },
        RecyclingGuideService: {
          merge: true,
        },
        WasteForm: {
          merge: true,
        },
        WasteFamily: {
          merge: true,
        },
        Flow: {
          merge: true,
        },
        FlowColor: {
          merge: true,
        },
        DropOffMap: {
          merge: true,
        },
        Request: {
          merge: true,
        },
        RequestAggregate: {
          merge: true,
        },
      },
    }),
  });
}

export default client;
