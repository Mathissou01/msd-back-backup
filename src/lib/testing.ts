import { prettyDOM } from "@testing-library/react";

export function removeUnstableHtmlPropertiesForTesting(
  htmlElement: HTMLElement,
  attributeName: string,
) {
  const domHTML = prettyDOM(htmlElement, Infinity).toString();
  if (!domHTML) return undefined;
  const regex = new RegExp(`${attributeName}(.*)"(.*)"`, "g");
  return domHTML.replaceAll(regex, "");
}
