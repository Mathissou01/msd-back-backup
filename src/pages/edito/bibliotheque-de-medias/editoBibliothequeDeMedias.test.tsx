import { render, screen } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import React from "react";
import { defaultMockData } from "../../../../__mocks__/mediaCreateFolderButtonMockData";
import EditoBibliothequeDeMedia from "./index.page";

describe("MediaCreateFolderButton", () => {
  it("check it renders correctly the component ", () => {
    const { container } = render(
      <MockedProvider mocks={defaultMockData}>
        <EditoBibliothequeDeMedia />
      </MockedProvider>,
    );

    const mediaCreateFolderButton = screen.getByTestId(
      "media-create-folder-button",
    );
    expect(mediaCreateFolderButton).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
