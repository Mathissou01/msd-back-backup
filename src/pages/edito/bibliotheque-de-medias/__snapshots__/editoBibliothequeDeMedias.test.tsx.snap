// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaCreateFolderButton check it renders correctly the component  1`] = `
<div>
  <hgroup
    class="c-PageTitle"
  >
    <h1
      data-testid="title"
    >
      Ajouter des m√©dias
    </h1>
    <p
      class="c-PageTitle__Description"
      data-testid="description"
    />
  </hgroup>
  <div
    class="c-EditoBibliothequeDeMedia__Options"
  >
    <div
      class="c-EditoBibliothequeDeMedia__Actions"
    >
      <div
        class="c-EditoBibliothequeDeMedia__Navigation"
      />
      <div
        class="c-EditoBibliothequeDeMedia__ActionButtons"
      >
        <div
          data-testid="media-create-folder-button"
        >
          <button
            class="c-CommonButton c-CommonButton_secondary c-CommonButton_fontSmall"
            type="button"
          >
            <div
              class="c-CommonButton__Picto c-CommonButton__Picto_add"
            />
            Ajouter un dossier
          </button>
        </div>
      </div>
    </div>
    <div
      class="c-EditoBibliothequeDeMedia__Filters"
    />
  </div>
  <div
    class="c-EditoBibliothequeDeMedia__Folders"
  >
    <h2>
      Dossiers
    </h2>
    <div
      class="c-EditoBibliothequeDeMedia__FolderCards"
    />
  </div>
  <div
    class="c-EditoBibliothequeDeMedia__MediaList"
  />
</div>
`;
